### PR TITLE
Removed several PDA checks for wasting less CUs

### PR DESCRIPTION
**Pull request by real-felix**
Several PDA checks were redundant, and are now removed. It fixes a bug causing a transfer to fail if the accounts caused the PDA checks to exceed the CU.

Also, now the owner transfer correctly moves the program authority to the new owner.

### CODE REVIEW

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs [hunk 1]
```diff
@@ -370,3 +378,12 @@ fn redeem_token(
 fn denormalize_dropoff_to_lamports(amount: u32) -> u64 {
     u64::from(amount) * 1_000
 }
+
+fn read_token_amount(token_account: &UncheckedAccount) -> Result<u64> {
+    msg!("TRY DESER");
```
Comment (nonergodic) [inline solana/programs/token-bridge-relayer/src/processor/inbound.rs:383]: Seems like a leftover

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs [hunk 2]
```diff
@@ -370,3 +378,12 @@ fn redeem_token(
...
+    let mut data: &[u8] = &token_account.try_borrow_data()?;
+    let data = TokenAccount::try_deserialize(&mut data)?;
+    msg!("DESER OK");
```
Comment (nonergodic) [inline solana/programs/token-bridge-relayer/src/processor/inbound.rs:386]: likewise

File: solana/tests/utils/helpers.ts [hunk 3]
```diff
@@ -166,6 +167,15 @@ export class TestsHelper {
     return sendAndConfirm(this.connection, ixs, payer, ...signers);
   }
 
+  async getTransaction(
+    signature: TransactionSignature,
+  ): Promise<VersionedTransactionResponse | null> {
+    return this.connection.getTransaction(signature, {
+      commitment: 'confirmed',
```
Comment (nonergodic) [inline solana/tests/utils/helpers.ts:174]: Hardcoded commitment levels are almost always a no-no. Why not rely on the default commitment level of the connection?

File: solana/tests/utils/tbr-wrapper.ts [hunk 4]
```diff
@@ -81,99 +81,162 @@ export class TbrWrapper {
     owner: PublicKey;
     feeRecipient: PublicKey;
     admins: PublicKey[];
-  }): Promise<TransactionSignature> {
-    return $.sendAndConfirm(await this.client.initialize(args), this.signer);
+  }): Promise<VersionedTransactionResponse | null> {
+    return $.getTransaction(
+      await $.sendAndConfirm(await this.client.initialize(args), this.signer),
+    );
```
Comment (nonergodic) [inline solana/tests/utils/tbr-wrapper.ts:87]: This common `getTransaction(sendAndConfirm( ..., this.signer))` call structure should preferably go into private function that just takes a promise to cut down on the boiler-plate.

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs [hunk 5]
```diff
@@ -160,6 +155,22 @@ pub fn complete_transfer(mut ctx: Context<CompleteTransfer>) -> Result<()> {
...
+        payer: ctx.accounts.payer.to_account_info(),
+        account: ctx.accounts.temporary_account.to_account_info(),
+        mint: ctx.accounts.mint.to_account_info(),
+        authority: ctx.accounts.wormhole_redeemer.to_account_info(),
+    }
+    .run_with_seeds(&[
+        SEED_PREFIX_TEMPORARY,
+        ctx.accounts.mint.to_account_info().key.as_ref(),
+        &[temporary_account_bump],
+    ])?;
```
Comment (nonergodic) [inline solana/programs/token-bridge-relayer/src/processor/inbound.rs:172]: What's the difference between this and just using the `init` in the context? You could pass the `temporary_account_bump` in there as instruction data too.

**Review by nonergodic — CHANGES_REQUESTED** [review]

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs (hunk 5 above)
Comment (real-felix) [reply solana/programs/token-bridge-relayer/src/processor/inbound.rs:172]: I put a comment explaining this: Anchor forbids doing so, for "security reasons".

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs (hunk 1 above)
Comment (real-felix) [reply solana/programs/token-bridge-relayer/src/processor/inbound.rs:383]: Removed

File: solana/programs/token-bridge-relayer/src/processor/inbound.rs (hunk 2 above)
Comment (real-felix) [reply solana/programs/token-bridge-relayer/src/processor/inbound.rs:386]: Removed

File: solana/tests/utils/helpers.ts (hunk 3 above)
Comment (real-felix) [reply solana/tests/utils/helpers.ts:174]: I use the helper's finality

File: solana/tests/utils/tbr-wrapper.ts (hunk 4 above)
Comment (real-felix) [reply solana/tests/utils/tbr-wrapper.ts]: `getTransaction` now accepts a promise as well.

File: solana/programs/token-bridge-relayer/src/processor/owner.rs [hunk 6]
```diff
@@ -70,12 +75,37 @@ pub struct ConfirmOwnerTransfer<'info> {
...
+    #[account(address = bpf_loader_upgradeable::ID)]
+    pub bpf_loader_upgradeable: UncheckedAccount<'info>,
+
     pub system_program: Program<'info, System>,
 }
 
 pub fn confirm_owner_transfer_request(ctx: Context<ConfirmOwnerTransfer>) -> Result<()> {
     let tbr_config = &mut ctx.accounts.tbr_config;
 
+    // Change the program authority to the new owner:
```
Comment (nonergodic) [inline solana/programs/token-bridge-relayer/src/processor/owner.rs:95]: :+1:

File: solana/tests/utils/tbr-wrapper.ts [hunk 7]
```diff
@@ -81,99 +81,152 @@ export class TbrWrapper {
...
+    return $.getTransaction($.sendAndConfirm(await this.client.initialize(args), this.signer));
   }
 
-  async submitOwnerTransferRequest(newOwner: PublicKey): Promise<TransactionSignature> {
-    return $.sendAndConfirm(await this.client.submitOwnerTransferRequest(newOwner), this.signer);
+  async submitOwnerTransferRequest(
+    newOwner: PublicKey,
+  ): Promise<VersionedTransactionResponse | null> {
+    return $.getTransaction(
+      $.sendAndConfirm(await this.client.submitOwnerTransferRequest(newOwner), this.signer),
```
Comment (nonergodic) [inline solana/tests/utils/tbr-wrapper.ts:92]: All these are still very boiler-plate-y.

I'd replace all instances of `$.getTransaction($.sendAndConfirm(await somePromise, this.signer))` with just `sendConfirmGet(somePromise)`.

**Review by nonergodic — APPROVED** [review]
:+1: just 2 nits

**Pull request merged**